common/peerprovider/ringpopprovider/provider_test.go

```diff
@@ -132,3 +132,43 @@ func (c *TestRingpopCluster) FindHostByAddr(addr string) (HostInfo, bool) {
 	}
 	return HostInfo{}, false
 }
+
+func TestLabelToPort(t *testing.T) {
+	tests := []struct {
```

Comment:
nit: if you're looking for the full set of tests, you also need to consider prefixes if we want to use `strconv.ParseInt(label, base=0, 16)` (base=0 is tricky according to doc, it is auto-detection of octal/dec/hex/...).
Otherwise consider switching to more explicit base=10 like `net.ParseAddrPort` does: 
`strconv.ParseUint(port, 10, 16)`

Reply:
good point. it defaults to 10 in this case but we should explicitly set to 10.